lib/rucio/common/policy.py

```diff
@@ -39,6 +51,107 @@
                     format='%(asctime)s\t%(process)d\t%(levelname)s\t%(message)s')
 
 
+def construct_surl_DQ2(dsn, filename):
+    """
+    Defines relative SURL for new replicas. This method
+    contains DQ2 convention. To be used for non-deterministic sites.
+    Method imported from DQ2.
+
+    @return: relative SURL for new replica.
+    @rtype: str
+    """
+    # check how many dots in dsn
+    fields = dsn.split('.')
+    nfields = len(fields)
+
+    if nfields == 0:
+        return '/other/other/%s' % (filename)
+    elif nfields == 1:
+        stripped_dsn = __strip_dsn(dsn)
+        return '/other/%s/%s' % (stripped_dsn, filename)
+    elif nfields == 2:
+        project = fields[0]
+        stripped_dsn = __strip_dsn(dsn)
+        return '/%s/%s/%s' % (project, stripped_dsn, filename)
+    elif nfields < 5 or re.match('user*|group*', fields[0]):
+        project = fields[0]
+        stripped_dsn = __strip_dsn(dsn)
+        return '/%s/%s/%s/%s/%s' % (project, fields[1], fields[2], stripped_dsn, filename)
+    else:
+        project = fields[0]
+        dataset_type = fields[4]
+        if nfields == 5:
+            tag = 'other'
+        else:
+            tag = __strip_tag(fields[-1])
+        stripped_dsn = __strip_dsn(dsn)
+        return '/%s/%s/%s/%s/%s' % (project, dataset_type, tag, stripped_dsn, filename)
+
+
+def construct_surl_T0(dsn, filename):
+    """
+    Defines relative SURL for new replicas. This method
+    contains Tier0 convention. To be used for non-deterministic sites.
+
+    @return: relative SURL for new replica.
+    @rtype: str
+    """
+    fields = dsn.split('.')
+    nfields = len(fields)
+    if nfields >= 3:
+        return '/%s/%s/%s/%s/%s' % (fields[0], fields[2], fields[1], dsn, filename)
+    elif nfields == 1:
+        return '/%s/%s/%s/%s/%s' % (fields[0], 'other', 'other', dsn, filename)
+    elif nfields == 2:
+        return '/%s/%s/%s/%s/%s' % (fields[0], fields[2], 'other', dsn, filename)
+    elif nfields == 0:
+        return '/other/other/other/other/%s' % (filename)
+
+
+def construct_surl(dsn, filename, naming_convention=None):
```

Comment:
This part, we should actually make configurable similar to the pfn2lfn algorithm. E.g. for Belle-2 we made some hardcoded changes to allow their naming convention for non-deterministic RSEs.

Reply:
(Not right now, I mean just in the medium term)